hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java

```diff
@@ -92,6 +99,39 @@ public XmlClientConfigBuilder() {
         this.in = locator.getIn();
     }
 
+    @Override
+    protected Document parse(InputStream inputStream) throws Exception {
+
+        final DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        Document doc;
+        try {
+            doc = builder.parse(inputStream);
+        } catch (final Exception e) {
+            String msg = "Failed to parse Config Stream"
+                    + "\nException: " + e.getMessage()
+                    + "\nHazelcastClient startup interrupted.";
+            LOGGER.severe(msg);
+            throw e;
+        } finally {
+            IOUtil.closeResource(inputStream);
+        }
+        return doc;
+    }
+
```

Comment:
Just if you did this because of the `finally`: A `finally` block will be executed even if you return within the `try-catch`-block (you can even overwrite the return value in a `finally` block, which is kinda creepy). So don't hesitate to return from a `try-catch`, `finally` always wins.